sjsonnet/src/sjsonnet/Std.scala

```diff
@@ -1019,4 +1013,34 @@ object Std {
     val output = str.toSeq.sliding(1).toList
     Val.Arr(output.map(s => Val.Lazy(Val.Str(s.toString()))).toSeq)
   }
+  
+  def getVisibleKeys(ev: EvalScope, v1: Val.Obj): Seq[String] = {
+    val keys = v1.getVisibleKeys()
+      .collect{case (k, false) => k}
+      .toSeq
+    
+    maybeSortKeys(ev, keys)
+  }
+  
+  def getAllKeys(ev: EvalScope, v1: Val.Obj): Seq[String] = {
+    val keys = v1.getVisibleKeys()
+      .collect{case (k, _) => k}
+      .toSeq
+    
+    maybeSortKeys(ev, keys)
+  }
+  
+  def maybeSortKeys(ev: EvalScope, keys: Seq[String]): Seq[String] = {
+    if(ev.preserveOrder) {
+      keys
+    } else {
+      keys.sorted
+    }
+  }
+  
+  def getObjValuesFromKeys(ev: EvalScope, fs: FileScope, v1: Val.Obj, keys: Seq[String]): Val.Arr = {
+    Val.Arr(keys.map { k =>
+      Val.Lazy(v1.value(k, -1)(fs, ev))
```

Comment:
I was a bit unsure about the `offset` argument here, but saw `-1` used elsewhere. Let me know if it should be different.